rpy2/rinterface.py

```diff
@@ -1097,26 +1098,149 @@ def _post_initr_setup() -> None:
     _update_R_ENC_PY()
 
 
+def _find_first(nodes, of_type):
+    for node in nodes:
+        if isinstance(node, of_type):
+            return node
+    raise ValueError(
+        f'No node of type {of_type}'
+    )
+
+
 def rternalize(function: typing.Callable) -> SexpClosure:
     """ Make a Python function callable from R.
 
     Takes an arbitrary Python function and wrap it in such a way that
     it can be called from the R side.
 
+    The Python ellipsis arguments`*args` and `**kwargs` are
+    translated to the R ellipsis `...`.
+
+    For example:
+
+    .. code-block:: python
+       @rternalize
+       def foo(x, *args, y=2):
+           pass
+
+    will be visible in R with the signature:
+
+    .. code-block:: r
+       function (x, ..., z)
+
+    The only limitation is that whenever `*args` and `**kwargs` are
+    both present in the Python declaration they must be consecutive.
+    For example:
+
+    .. code-block:: python
+       def foo(x, *args, y=2, **kwargs):
+           pass
+
+    is a valid Python declaration. However, it cannot be "rternalized"
+    because the R ellipsis can only appear at most once in the signature
+    of a given function. Trying to apply the decorator `rternalize` would
+    raise an exception.
+
+    The following Python function can be "rternalized":
+
+    .. code-block:: python
+       def foo(x, *args, **kwargs):
+           pass
+
+    It is visible to R with the signature
+
+    .. code-block:: r
+       function (x, ..., z)
```

Comment:
What is `z`?

Reply:
Good question. This is a mistake. The R signature will be `function (x, ...)`.